.github/workflows/release.yml

```diff
@@ -3,60 +3,109 @@ name: Release
+      archive-name: ${{ steps.archive.outputs.name }}
     steps:
-      - name: Check out repository
+      - name: "Checkout ${{ github.event.repository.name }}@${{ needs.release-prep.outputs.release-branch }}"
         uses: actions/checkout@v4
         with:
+          ref: ${{ needs.release-prep.outputs.release-branch }}
           persist-credentials: false
-          ref: "${{ inputs.ref }}"
 
-      - name: Setup `hatch`
+      - name: "Setup `hatch`"
         uses: dbt-labs/dbt-adapters/.github/actions/setup-hatch@main
 
-      - name: Inputs
-        id: release-inputs
+      - name: "Set archive name"
+        id: archive
         run: |
-          version=$(hatch version)
-          archive_name=dbt-postgres-$version-${{ inputs.deploy-to }}
-          echo "archive-name=$archive_name" >> $GITHUB_OUTPUT
+          archive_name=${{ github.event.repository.name }}-${{ inputs.version }}-${{ inputs.deploy-to }}
+          echo "name=$archive_name" >> $GITHUB_OUTPUT
 
-      - name: Build `dbt-postgres`
+      - name: "Build ${{ github.event.repository.name }}"
         uses: dbt-labs/dbt-adapters/.github/actions/build-hatch@main
         with:
-          archive-name: ${{ steps.release-inputs.outputs.archive-name }}
+          archive-name: ${{ steps.archive.outputs.name }}
 
-      - name: Publish to PyPI
+  pypi-release:
+    name: "PyPI release"
+    if: ${{ !failure() && !cancelled() && !inputs.only_docker }}
+    runs-on: ubuntu-latest
+    needs: build-release
+    environment:
+      name: ${{ inputs.deploy-to }}
+      url: ${{ vars.PYPI_PROJECT_URL }}
+    permissions:
+      id-token: write  # this permission is required for trusted publishing
+    steps:
+      - name: "Publish to PyPI"
         uses: dbt-labs/dbt-adapters/.github/actions/publish-pypi@main
         with:
-          pypi-repository-url: ${{ vars.PYPI_REPOSITORY_URL }}
-          archive-name: ${{ steps.release-inputs.outputs.archive-name }}
+          repository-url: ${{ vars.PYPI_REPOSITORY_URL }}
+          archive-name: ${{ needs.build-release.outputs.archive-name }}
+
+  github-release:
+    name: "GitHub release"
+    if: ${{ !failure() && !cancelled() && !inputs.only_docker }}
+    needs:
+      - build-release
+      - release-prep
+    uses: dbt-labs/dbt-adapters/.github/workflows/github-release.yml@main
+    with:
+      sha: ${{ needs.release-prep.outputs.release-sha }}
+      version_number: ${{ inputs.version }}
+      changelog_path: ${{ needs.release-prep.outputs.changelog-path }}
+      test_run: ${{ inputs.deploy-to == 'test' }}
+      archive_name: ${{ needs.build-release.outputs.archive-name }}
+
+  docker-release:
+    name: "Docker release"
+    # We cannot release to docker on a test run because it uses the tag in GitHub as
+    # what we need to release but draft releases don't actually tag the commit so it
+    # finds nothing to release
+    if: ${{ !failure() && !cancelled() && (inputs.deploy-to == 'prod' || inputs.only_docker) }}
```

Comment:
Doesn't this logic mean you can specify `test` and `only_docker` and it gets out the door? Feel like that shouldn't be an allowed input state.

Reply:
Passing `test_run=true` into the reusable workflow, which is what happens in the above scenario, means that the Docker container will only be built, verifying that it's a valid image; it will not push the built image to the container registry. The argument gets passed into `push`, which is referenced [here](https://github.com/docker/build-push-action?tab=readme-ov-file#inputs).